practicesharp/Numeric_Base.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace basic_calculator
+{
+    public interface Numeric_Base
+    {
+        private static string field;
+        public bool Validate(string number);
+        public double To_Decimal(string number);
+        public void print(double number);
+    }
+
+
+    class Binary : Numeric_Base
+    {
+        private static string field = "01";
+        private short weight = 2;
+        public bool Validate(string number)
+        {
+            foreach (char letter in number)
+                if (!field.Contains(letter)) return false;
+            return true;
+        }
+        public double To_Decimal(string number)
+        {
+            double value = 0;
+            foreach (char letter in number)
+            {
+                value *= weight;
+                value += (letter - '0');
+            }
+            return value;
+        }
+        public void print(double number)
```

Comment:
What if you want to write the number to a file?
This is a violation of single responsibility.
Remove the function from this class.